diff --git a/tensorflow/lite/kernels/register.cc b/tensorflow/lite/kernels/register.cc
index c09525ff74..76a7cd085f 100644
--- a/tensorflow/lite/kernels/register.cc
+++ b/tensorflow/lite/kernels/register.cc
@@ -20,9 +20,9 @@ namespace ops {
 
 namespace custom {
 
-TfLiteRegistration* Register_AUDIO_SPECTROGRAM();
-TfLiteRegistration* Register_MFCC();
-TfLiteRegistration* Register_DETECTION_POSTPROCESS();
+//TfLiteRegistration* Register_AUDIO_SPECTROGRAM();
+//TfLiteRegistration* Register_MFCC();
+//TfLiteRegistration* Register_DETECTION_POSTPROCESS();
 TfLiteRegistration* Register_IF();
 TfLiteRegistration* Register_WHILE();
 
@@ -144,8 +144,8 @@ TfLiteRegistration* Register_MATRIX_DIAG();
 TfLiteRegistration* Register_QUANTIZE();
 TfLiteRegistration* Register_MATRIX_SET_DIAG();
 TfLiteRegistration* Register_HARD_SWISH();
-TfLiteRegistration* Register_NON_MAX_SUPPRESSION_V4();
-TfLiteRegistration* Register_NON_MAX_SUPPRESSION_V5();
+//TfLiteRegistration* Register_NON_MAX_SUPPRESSION_V4();
+//TfLiteRegistration* Register_NON_MAX_SUPPRESSION_V5();
 
 const TfLiteRegistration* BuiltinOpResolver::FindOp(tflite::BuiltinOperator op,
                                                     int version) const {
@@ -384,10 +384,10 @@ BuiltinOpResolver::BuiltinOpResolver() {
              /* min_version */ 1,
              /* max_version */ 2);
   AddBuiltin(BuiltinOperator_MATRIX_SET_DIAG, Register_MATRIX_SET_DIAG());
-  AddBuiltin(BuiltinOperator_NON_MAX_SUPPRESSION_V4,
-             Register_NON_MAX_SUPPRESSION_V4());
-  AddBuiltin(BuiltinOperator_NON_MAX_SUPPRESSION_V5,
-             Register_NON_MAX_SUPPRESSION_V5());
+  //AddBuiltin(BuiltinOperator_NON_MAX_SUPPRESSION_V4,
+  //           Register_NON_MAX_SUPPRESSION_V4());
+  //AddBuiltin(BuiltinOperator_NON_MAX_SUPPRESSION_V5,
+  //           Register_NON_MAX_SUPPRESSION_V5());
 
   // WARNING: Control flow ops are experimental and subject to change.
   AddBuiltin(BuiltinOperator_IF, tflite::ops::custom::Register_IF());
@@ -395,11 +395,11 @@ BuiltinOpResolver::BuiltinOpResolver() {
 
   // TODO(andrewharp, ahentz): Move these somewhere more appropriate so that
   // custom ops aren't always included by default.
-  AddCustom("Mfcc", tflite::ops::custom::Register_MFCC());
-  AddCustom("AudioSpectrogram",
-            tflite::ops::custom::Register_AUDIO_SPECTROGRAM());
-  AddCustom("TFLite_Detection_PostProcess",
-            tflite::ops::custom::Register_DETECTION_POSTPROCESS());
+  //AddCustom("Mfcc", tflite::ops::custom::Register_MFCC());
+  //AddCustom("AudioSpectrogram",
+  //          tflite::ops::custom::Register_AUDIO_SPECTROGRAM());
+  //AddCustom("TFLite_Detection_PostProcess",
+  //          tflite::ops::custom::Register_DETECTION_POSTPROCESS());
 }
 
 }  // namespace builtin
